src/Connections.Abstractions/Features/IConnectionLifetimeNotificationFeature.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Connections.Features
+{
+    public interface IConnectionLifetimeNotificationFeature
+    {
+        CancellationToken ConnectionClosing { get; set; }
```

Comment:
Why is this better than ApplicationStopping?

Reply:
I don't want to plumb IApplicationLifetime through to every protocol (even though we have one now). This seems to fit more nicely with our existing feature model.